apps/rush-lib/src/logic/pnpm/PnpmLinkManager.ts

```diff
@@ -222,11 +222,17 @@ export class PnpmLinkManager extends BaseLinkManager {
     //   C%3A%2Fwbt%2Fcommon%2Ftemp%2Fprojects%2Fapi-documenter.tgz
     //   C%3A%2Fdev%2Fimodeljs%2Fimodeljs%2Fcommon%2Ftemp%2Fprojects%2Fpresentation-integration-tests.tgz_jsdom@11.12.0
     //   C%3A%2Fdev%2Fimodeljs%2Fimodeljs%2Fcommon%2Ftemp%2Fprojects%2Fbuild-tools.tgz_2a665c89609864b4e75bc5365d7f8f56
-    const folderNameInLocalInstallationRoot: string =
-      uriEncode(Text.replaceAll(absolutePathToTgzFile, path.sep, '/')) + folderNameSuffix;
+    let folderNameInLocalInstallationRoot: string =
+      uriEncode(Path.convertToSlashes(absolutePathToTgzFile)) + folderNameSuffix;
+
+    // PNPM 6 changed formatting to replace all special chars with '+'
```

Comment:
Can you cite a source for this?  Looks like maybe it is performed by [dependency-path/src/index.ts](https://github.com/pnpm/pnpm/blob/9ceab68f06eeac185b8b2712a3c8ec0fb26a74ef/packages/dependency-path/src/index.ts#L134).

If so, that code seems to be only replacing `:` and `/` -- not "all special chars".  (?)

Reply:
Looks like this is correct, good catch! I'll update